src/consensus/utils.rs

```diff
@@ -9,14 +9,20 @@ pub async fn add_peer_to_toc_and_consensus_state(
     toc: &Arc<TableOfContent>,
     peer_id: PeerId,
     uri: Uri,
+    add_remote_replicas: bool,
 ) -> Result<(PeerId, Vec<(PeerId, String)>), ConsensusError> {
-    let (peer_id, all_peers) = consensus_state.add_peer(peer_id, uri).await.map_err(|e| {
-        ConsensusError::ServiceError(format!("Failed to add peer to local consensus state: {e}"))
-    })?;
+    let (_leader_peer_id, all_peers) =
+        consensus_state.add_peer(peer_id, uri).await.map_err(|e| {
+            ConsensusError::ServiceError(format!(
+                "Failed to add peer to local consensus state: {e}"
+            ))
+        })?;
 
-    let collections = toc.collections.read().await;
-    for collection in collections.values() {
-        collection.add_remote_replicas(peer_id).await;
+    if add_remote_replicas {
+        let collections = toc.collections.read().await;
+        for collection in collections.values() {
+            collection.add_remote_replicas(peer_id).await;
+        }
     }
```

Comment:
The problem was here. I was resetting `peer_id` as the returned value from the function which always returned the leader peer ID.